chore: add optional coder_app to faq (#11351)
coder/coder: docs/faqs.md

Colima will show the path to the docker socket so I have a
[Coder template](./docker-code-server/main.tf) that prompts the Coder admin to
enter the docker socket as a Terraform variable.

## How to make a `coder_app` optional?

An example use case is the user should decide if they want a browser-based IDE
like code-server when creating the workspace.

> 1. Create a `coder_parameter` as a `bool` to ask the user if they want the

Reviewer: ```suggestion
1. Add a `coder_parameter` as a `bool` to ask the user if they want the
```

Reviewer 2: changed

   code-server IDE

```hcl
data "coder_parameter" "code_server" {

>   name        = "code-server (optional)"
>   description = "Use VS Code in a browser"

Reviewer: ```suggestion
  name        = "Do you want code-server in your workspace?"
  description = "Use VS Code in a browser."
```

  type        = "bool"
  default     = false
  mutable     = true
  icon        = "/icon/code.svg"
  order       = 6
}
```

2. Add conditional logic to the `startup_script` to install and start

>    code-server if the `bool` is true

Reviewer: ```suggestion
   code-server depending on the value of the added `coder_parameter`.
```

```sh

> # install and code-server, VS Code in a browser

Reviewer: ```suggestion
# install and start code-server, VS Code in a browser
```

if [ ${data.coder_parameter.code_server.value} = true ]; then
  echo "🧑🏼‍💻 Downloading and installing the latest code-server IDE..."
  curl -fsSL https://code-server.dev/install.sh | sh
  code-server --auth none --port 13337 >/dev/null 2>&1 &
fi
```

> 3. Add a Terraform meta-argument `count` in the `coder_app` resource so it will
>    only create the resource if the `coder_parameter` is true

Reviewer: ```suggestion
3. Add a Terraform meta-argument [`count`](https://developer.hashicorp.com/terraform/language/meta-arguments/count) in the `coder_app` resource so it will
   only create the resource if the `coder_parameter` is `true`
```

```hcl
# code-server
resource "coder_app" "code-server" {

>   count        = data.coder_parameter.code_server.value ? 1 : 0

Reviewer: ```suggestion
  count         = data.coder_parameter.code_server.value ? 1 : 0
```
